src/value/serde_yaml.rs

```diff
@@ -12,20 +12,18 @@ impl Value for SerdeYaml {
     type Item = SerdeYaml;
     fn items<'a>(&'a self) -> Option<Box<dyn Iterator<Item = (Self::Key, &'a Self::Item)> + 'a>> {
         match *self {
-            SerdeYaml::String(_) | SerdeYaml::Number(_) | SerdeYaml::Bool(_) | SerdeYaml::Null => {
-                None
-            }
+            SerdeYaml::String(_)
+            | SerdeYaml::Number(_)
+            | SerdeYaml::Bool(_)
+            | SerdeYaml::Null
+            | SerdeYaml::Tagged(_) => None,
             SerdeYaml::Sequence(ref inner) => Some(Box::new(
                 inner.iter().enumerate().map(|(i, v)| (Key::Index(i), v)),
             )),
             SerdeYaml::Mapping(ref inner) => Some(Box::new(inner.iter().map(|(k, v)| {
-                (
-                    Key::String(
-                        to_string(k).expect("yaml value to serialize into yaml correctly")[4..]
-                            .to_owned(),
-                    ),
-                    v,
-                )
+                let k = to_string(k).expect("yaml value to serialize into yaml correctly");
+                let k = &k[..k.len() - 1]; // trim trailing carriage return
```

Comment:
Can you elaborate why this assumption was added? Particularly, how can we be sure this is truly always the case.

Reply:
I had to modify the `k[4..]` while updating `treediff` to `serde_yaml` 0.9 because the output string produce by `to_string()` is not including any padding anymore.

Then I found, as mentioned in 6d524b42f25822846f45bcd345d400c494b387ba, that the `\n` in the key value is not correct for a string key. I've implemented a [function in my project that retrieve a `Value` based on the keys provided by `treediff`](https://github.com/glehmann/yage/blob/3adb7223af7b0450de460d44efa87a7156d26cfd/src/edit.rs#L70), and I had to remove the trailing `\n` in my code to make it work. So I thought it might be good to fix it too.

But now that I better understand the code, I think I should have used `serde_yaml::from_str()`  in my code in order to support key types that are not strings. In that case the `\n` is not much problematic, it's just cosmetic. I'll modify the PR to check if there is a trailing `\n` before removing it, so we can be sure it will always work.